samcli/lib/deploy/deployer.py

```diff
@@ -336,10 +336,14 @@ def describe_stack_events(self, stack_name, time_stamp_marker, **kwargs):
                 paginator = self._client.get_paginator("describe_stack_events")
                 response_iterator = paginator.paginate(StackName=stack_name)
                 stack_status = describe_stacks_resp["Stacks"][0]["StackStatus"]
+                latest_time_stamp_marker = time_stamp_marker
                 for event_items in response_iterator:
                     for event in event_items["StackEvents"]:
                         if event["EventId"] not in events and utc_to_timestamp(event["Timestamp"]) > time_stamp_marker:
                             events.add(event["EventId"])
+                            latest_time_stamp_marker = max(
```

Comment:
I understand the logic here, could you add a unit test to showcase this?

Reply:
Thank you for the review!
I've added the unit test. Could you review it?